cumulusci/core/debug.py

```diff
@@ -0,0 +1,32 @@
+from contextlib import contextmanager
+from contextvars import ContextVar
+
+_DEBUG_MODE = ContextVar("debug_mode", default=False)
+
+
+# This may have levels/modes/flags someday
+# For now let's discourage anyone from using "DebugMode is True"
+class DebugMode:
+    """A singleton/proxy that behaves truthy
+    when debug mode is not enabled in a context
+    and falsy otherwise.
+
+    Debug mode can be set for a context with the
+    context manager called `set`."""
+
+    def __bool__(self):
+        return _DEBUG_MODE.get()
+
+    @contextmanager
+    def set(self, enable: bool):
+        token = _DEBUG_MODE.set(enable)
+        try:
+            yield
+        finally:
+            _DEBUG_MODE.reset(token)
+
+
+DebugMode = DebugMode()
+
```

Comment:
I think I would have just written `set_debug_mode` and `get_debug_mode` functions rather than a singleton class, but I guess that's more a matter of taste and don't really have a substantive critique of the class.

A couple minor things though:
- It took me a minute to figure out the intended way to set and get the debug mode. Would be helpful to have examples in the docstring.
- It irks me that the instance of the singleton is written in camel case, which usually signifies a class in Python

Reply:
I didn't feel as strongly about it as you did, so I've changed to a function-based syntax in 28aea0b9